autorest/codegen/models/constant_schema.py

```diff
@@ -84,6 +84,9 @@ def from_yaml(cls, namespace: str, yaml_data: Dict[str, Any], **kwargs) -> "Cons
         )
 
     def get_json_template_representation(self, **kwargs: Any) -> Any:
+        # for better display effect, use constant value instead of var type
```

Comment:
I don't think you need the if statement here, just set the default value declaration to `self.value`